core/controllers/reader.py

```diff
@@ -213,6 +213,24 @@ def get(self, exploration_id):
 
         exploration_rights = rights_manager.get_exploration_rights(
             exploration_id, strict=False)
+
+        # Retrieve all classifiers for the exploration.
+        state_classifier_mapping = {}
+        classifier_training_jobs = (
+            classifier_services.get_classifier_training_jobs(
+                exploration_id, exploration.version, exploration.states))
+        for index, state_name in enumerate(exploration.states):
+            if classifier_training_jobs[index] is None:
+                state_classifier_mapping[state_name] = None
+            else:
+                classifier_data = classifier_training_jobs[
+                    index].classifier_data
+                algorithm_id = classifier_training_jobs[index].algorithm_id
+                state_classifier_mapping[state_name] = {
+                    'algorithm_id': algorithm_id,
+                    'classifier_data': classifier_data
```

Comment:
@seanlip It's necessity comes down to this. Lets say the classifier for TextInput has a data_schema_version increment. This in turn changes the TextClassifierPredictionService.predict() method.
So, we can either maintain each data_schema_version's corresponding prediction service or just make the prediction service use the latest schema. 
I am slightly inclined towards the first option when I think about it. If we do end up doing that, we'll pass data_schema_version in this as well (and modify RegistryService accordingly).